### PR TITLE
Reduce access level in classes and methods in IoT hub service client

### PR DESCRIPTION
**Pull request by timtay-microsoft**
And replace all the basic handwritten getters/setters with lombok annotations for simplicity

### CODE REVIEW

File: service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/ExportImportDevice.java [hunk 1]
```diff
@@ -258,84 +147,6 @@ public int hashCode()
...
     }
 
-    /**
-     * Retrieves information from the provided parser and returns it in a new ExportImportDevice instance. All information on this shall be overwritten.
-     * @param parser the parser to read from
-     * @throws IllegalArgumentException if the provided parser is missing the authentication or id fields. It also shall
-     * be thrown if the authentication object in the parser uses SAS authentication and is missing one of the symmetric key fields, or if it uses SelfSigned authentication
-     * and is missing one of the thumbprint fields.
-     */
-    ExportImportDevice(ExportImportDeviceParser parser) throws IllegalArgumentException
```
Comment (timtay-microsoft) [inline service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/ExportImportDevice.java:268]: unused constructor

File: service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/ServiceStatistics.java [hunk 2]
```diff
@@ -1,25 +0,0 @@
...
- * Licensed under the MIT license. See LICENSE file in the project root for full license information.
- */
-
-package com.microsoft.azure.sdk.iot.service;
-
-/**
- * Data structure for service statistics received in Json array
- * Provide getters and setters for statistics properties
- */
-public class ServiceStatistics
```
Comment (timtay-microsoft) [inline service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/ServiceStatistics.java:12]: Unused class

File: service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/BaseDevice.java [hunk 3]
```diff
@@ -60,7 +61,7 @@ protected BaseDevice(String deviceId, SymmetricKey symmetricKey)
      * @param deviceId Name of the device (used as device id)
      * @param authenticationType - The type of authentication used by this device.
      */
-    protected BaseDevice(String deviceId, AuthenticationType authenticationType)
+    BaseDevice(String deviceId, AuthenticationType authenticationType)
```
Comment (vinagesh) [inline service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/BaseDevice.java:64]: Does this need the private keyword like other places?

File: service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/FeedbackReceiver.java [hunk 4]
```diff
@@ -39,9 +39,9 @@
      * @param iotHubServiceClientProtocol protocol to be used
      *
      */
-    public FeedbackReceiver(String hostName, String userName, String sasToken, IotHubServiceClientProtocol iotHubServiceClientProtocol)
+    FeedbackReceiver(String hostName, String userName, String sasToken, IotHubServiceClientProtocol iotHubServiceClientProtocol)
```
Comment (vinagesh) [inline service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/FeedbackReceiver.java:42]: Add the private keyword?

File: service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/ServiceAuthenticationWithSharedAccessPolicyKey.java [hunk 5]
```diff
@@ -11,15 +11,15 @@
...
+class ServiceAuthenticationWithSharedAccessPolicyKey extends AuthenticationMethod
 {
     /**
      * Constructor to create instance from policy name and policy key
      *
      * @param policyName The policy name string
      * @param key The policy key string
      */
-    public ServiceAuthenticationWithSharedAccessPolicyKey(String policyName, String key)
+    ServiceAuthenticationWithSharedAccessPolicyKey(String policyName, String key)
```
Comment (vinagesh) [inline service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/ServiceAuthenticationWithSharedAccessPolicyKey.java:22]: private?

File: service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/ServiceAuthenticationWithSharedAccessPolicyToken.java [hunk 6]
```diff
@@ -47,46 +47,9 @@ public IotHubConnectionString populate(IotHubConnectionString iotHubConnectionSt
      * @param policyName The policy name string
      * @param token The token string
      */
-    public ServiceAuthenticationWithSharedAccessPolicyToken(String policyName, String token)
+    ServiceAuthenticationWithSharedAccessPolicyToken(String policyName, String token)
```
Comment (vinagesh) [inline service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/ServiceAuthenticationWithSharedAccessPolicyToken.java:50]: private?

**Review by vinagesh — APPROVED** [review]

File: service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/ServiceAuthenticationWithSharedAccessPolicyToken.java (hunk 6 above)
Comment (timtay-microsoft) [reply service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/ServiceAuthenticationWithSharedAccessPolicyToken.java:50]: This constructor is called from another class within this package, so I can't mark it as private. This lack of modifier makes the constructor "Package private" which means only classes within this package can access it. Classes outside of the package, even if they extend this class, can't access it

File: service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/ServiceAuthenticationWithSharedAccessPolicyKey.java (hunk 5 above)
Comment (timtay-microsoft) [reply service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/ServiceAuthenticationWithSharedAccessPolicyKey.java:22]: This constructor is called from another class within this package, so I can't mark it as private. This lack of modifier makes the constructor "Package private" which means only classes within this package can access it. Classes outside of the package, even if they extend this class, can't access it

File: service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/FeedbackReceiver.java (hunk 4 above)
Comment (timtay-microsoft) [reply service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/FeedbackReceiver.java:42]: This constructor is called from another class within this package, so I can't mark it as private. This lack of modifier makes the constructor "Package private" which means only classes within this package can access it. Classes outside of the package, even if they extend this class, can't access it

File: service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/BaseDevice.java (hunk 3 above)
Comment (timtay-microsoft) [reply service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/BaseDevice.java:64]: This constructor is called from another class within this package, so I can't mark it as private. This lack of modifier makes the constructor "Package private" which means only classes within this package can access it. Classes outside of the package, even if they extend this class, can't access it

**Pull request merged**